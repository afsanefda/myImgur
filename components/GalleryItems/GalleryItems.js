// @flow
import React from 'react';
import { useSelector } from 'react-redux';
import Link from 'next/link';
import {
  ItemWrapper, ItemWrapperImg, ItemWrapperVideo, ItemWrapperTitle,
} from './style';

type Props = {
  index: number,
};

const GalleryItems = ({ index }:Props) => {
  const items = useSelector((state) => state.galleryList.data);
  const coverSrc = items[index]?.images
    ? items[index]?.images.find((el) => el.id === items[index]?.cover)?.link
    : items[index]?.link;
  return (
    items[index]?.images ? (
      <ItemWrapper>
        <Link href={`/image/${items[index]?.id}`}>
          <a>
            {items[index]?.images[0].type === 'video/mp4' ? (
              <div>
                <ItemWrapperVideo autoPlay loop muted>
                  <source
                    type="video/mp4"
                    src={coverSrc}
                  />
                </ItemWrapperVideo>
              </div>
              ) : (
                <ItemWrapperImg
                  src={coverSrc}
                  alt={items[index]?.title}
                />
              )}
            <ItemWrapperTitle>{items[index]?.title}</ItemWrapperTitle>
          </a>
        </Link>
      </ItemWrapper>
    ) : (
      <ItemWrapper>
        <Link href={`/image/${items[index]?.id}`}>
          <a>
            <ItemWrapperImg src={coverSrc} alt={items[index]?.title} />
            <ItemWrapperTitle>{items[index]?.title}</ItemWrapperTitle>
          </a>
        </Link>
      </ItemWrapper>
    )
  );
};
export default GalleryItems;
