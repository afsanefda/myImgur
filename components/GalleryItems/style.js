import styled from 'styled-components';

export const ItemWrapperImg = styled.img`
  width: 100%;
  height: 180px;
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
`;

export const ItemWrapperVideo = styled.video`
  width: 100%;
  height: 180px;
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
`;

export const ItemWrapperTitle = styled.p`
  text-align: left;
  padding: 5px 10px;
  color: #A65215;
  font-size: 13px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  height: 38px;
  overflow: hidden;
`;

export const ItemWrapper = styled.div`
    flex: 1;
    text-align: center;
    font-size: 80%;
    box-shadow: 0 5px 6px -6px #777;
    background: #f5f3f3;
    border-radius: 6px;
    width: 100%;
    &:hover {
    box-shadow: 0 5px 6px -4px #212121;
    }
  }
`;
