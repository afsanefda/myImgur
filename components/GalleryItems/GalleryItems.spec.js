import React from 'react';
import createStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import GalleryItems from './GalleryItems';

describe('<GalleryItems />', () => {
  let rootElement;
  let testRenderer;
  let store;

  beforeEach(async () => {
    store = createStore([])({
      galleryList: {
        loading: false,
        data: [
          {
            account_id: 10047121,
            account_url: 'anlyin',
            ad_config: {},
            ad_type: 0,
            ad_url: '',
            comment_count: 120,
            cover: 'EegS2cp',
            cover_height: 854,
            cover_width: 684,
            datetime: 1598349693,
            description: null,
            downs: 21,
            favorite: false,
            favorite_count: 956,
            id: 'Znq9l9n',
            images: [{ id: 1, title: 'test', type: 'video/mp4' }, { id: 2, title: 'test2', type: 'img/png' }],
            images_count: 1,
            in_gallery: true,
            in_most_viral: true,
            include_album_ads: false,
            is_ad: false,
            is_album: true,
            layout: 'blog',
            link: 'https://imgur.com/a/Znq9l9n',
            nsfw: false,
            points: 3056,
            privacy: 'hidden',
            score: 3104,
            section: '',
            tags: (5)[{ name: 'test', accent: 'test' }],
            title: 'LEGO loom makes beautiful scarves',
            topic: 'No Topic',
            topic_id: 29,
            ups: 3077,
            views: 97375,
            vote: null,
          },
        ],
        error: false,
      },
    });
    store.dispatch = jest.fn();

    rootElement = () => (
      <Provider store={store}>
        <GalleryItems index={0} />
      </Provider>
    );

    testRenderer = renderer.create(rootElement());
  });

  it('Should find the correct index', () => {
    expect(testRenderer.root.props.children.props.index).toBe(0);
    expect(testRenderer.root.findAllByProps({ 'data-test-id': 'video-wrapper' }).length).toBe(1);
  });

  it('Should render correctly', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
