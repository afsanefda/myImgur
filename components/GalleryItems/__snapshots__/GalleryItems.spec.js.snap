// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryItems /> Should render correctly 1`] = `
<div
  className="style__ItemWrapper-sc-1nmiqb7-3 jXOryF"
>
  <a
    href="/image/Znq9l9n"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      data-test-id="video-wrapper"
    >
      <video
        autoPlay={true}
        className="style__ItemWrapperVideo-sc-1nmiqb7-1 djFQqb"
        loop={true}
        muted={true}
        preload="auto"
      >
        <source
          type="video/mp4"
        />
      </video>
    </div>
    <p
      className="style__ItemWrapperTitle-sc-1nmiqb7-2 jaKrQv"
    >
      LEGO loom makes beautiful scarves
    </p>
  </a>
</div>
`;
